arduino/i2c_slave_code_rccar/i2c_slave_rccar/i2c_slave_rccar.ino

```diff
@@ -89,14 +99,44 @@ bool checkRemote() {
 
 // called when data is sent through i2c to arduino
 // to update the motor values
-void receiveEvent() {
+void receiveEvent(int bytes) {
     // check incoming bytes from master
     uint8_t read_array[2];
-    while(Wire.available()) {
+    int i = 0;
+    for(int i = 0; Wire.available() && i < 2; i++) {
         read_array[i] = Wire.read();
     }
+    
     movement[0] = read_array[0]; // throttle value
     movement[1] = read_array[1]; // steering value
```

Comment:
I've only just spotted this, but there is the possibility that you'll read garbage values from ``read_array``, if ``Wire.available()`` returns ``false`` unexpectedly. I guess in that case, we'd want to leave ``movement`` as is?

Reply:
I have added a condition now to check whether all bytes were read successfully